File: python/linear_search.py
array_to_search_through = []
for number in range(1, 1001):
    array_to_search_through.append(number)

def linear_search(value_to_find, array_to_search_through):
  output = None
  for i, el in enumerate(array_to_search_through):
    if el == value_to_find:
      output = i
      break
  return output

def linear_search_global(value_to_find, array_to_search_through):
    return [idx for idx, el in enumerate(array_to_search_through) if el == value_to_find]

print(linear_search(24, [11,12,3,2]))
print(linear_search_global('d', list('bananas')))
print(linear_search_global('b', list('bananas')))
print(linear_search_global('a', list('bananas')))

File: js/linearSearch.js
// exports.

function isSearchValue(element, val) {
  return element === val;
}

exports.linearSearch = function(valueToFind, arrayToSearchThrough) {
  // account for undefined not in array
    let output = undefined;
    for (let i = 0; i < arrayToSearchThrough.length; i++){
        if (isSearchValue(arrayToSearchThrough[i], valueToFind)) {
          // i'm going to do something
          output = i;
          break;
        }
    }
  // move through the array
  // compare value to index position
  return output;
};



// console.log(linearSearch(4, [1,2,3]))
// console.log(linearSearch("A", ["C", "A", "T"]))
// console.log(linearSearch("A", "CAT"))

exports.linearSearchGlobal = function(valueToFind, arrayToSearchThrough) {
  let output = [];
  for (let i = 0; i < arrayToSearchThrough.length; i++){
      if (isSearchValue(arrayToSearchThrough[i], valueToFind)){
          output.push(i);
      }
  }
  return output;
  // variable for an array for index position
  //if statement 
}

// console.log(linearSearchGlobal("S", "BANANAS"));